pkg/kudoctl/resources/upgrade/operatorversion.go

```diff
@@ -0,0 +1,112 @@
+package upgrade
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	"github.com/thoas/go-funk"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/dependencies"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/resolver"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/resources/install"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/kudobuilder/kudo/pkg/util/convert"
+)
+
+// OperatorVersion upgrades an OperatorVersion and its Instance.
+// For the updated Instance, new parameters can be provided.
+func OperatorVersion(kc *kudo.Client, newOv *v1beta1.OperatorVersion, instanceName, namespace string, parameters map[string]string, resolver resolver.Resolver) error {
```

Comment:
This is `UpgradeOperatorVersion` that was in `pkg/kudoctl/util/kudo` with added dependency handling.

Reply:
This method has grown quite a bit. Should we attempt to clean it up? Separate some concerns like version checking or dependency installation?

Reply:
Yes, thought about this as well. Separate upgrading OV and updating the Instance with new OV and changes parameters.